### PR TITLE
Introduce Command - wrapper for closures used in Props for data-driven UI

### PR DESCRIPTION
**Pull request by igorkravchenko**
Add wrapper for closure to be used in `Props` of data-driven UI.
It makes closure conform `Equatable` and `Hashable`, providing info about callee.

MOB-1706

### CODE REVIEW

File: GliaWidgets/Lib/Command/Command.swift [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+    func execute(with value: T) {
+        closure(value)
+    }
+
+    func callAsFunction(_ value: T) {
+        execute(with: value)
+    }
+
+    static func == (lhs: Self, rhs: Self) -> Bool {
```
Comment (gersonnoboa) [inline GliaWidgets/Lib/Command/Command.swift:41]: Completely unrelated and of course nothing to do with your code, I have always despised this implementation by Apple. Who thought `lhs` and `rhs` were good names?

**Review by gersonnoboa — APPROVED** [review]

File: GliaWidgets/Lib/Command/Command.swift (hunk 1 above)
Comment (igorkravchenko) [reply GliaWidgets/Lib/Command/Command.swift:41]: I think it came from the languages like C/C++. LHS stands for left-hand side. I guess back in the day length of variables did matter. Also in math it is also commonly [used](https://en.wikipedia.org/wiki/Sides_of_an_equation).

File: GliaWidgets/Lib/Command/Command.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        self.function = "\(function)"
+        self.line = line
+        self.closure = closure
+    }
+
+    func execute(with value: T) {
+        closure(value)
+    }
+
+    func callAsFunction(_ value: T) {
```
Comment (EgorovEI) [inline GliaWidgets/Lib/Command/Command.swift:37]: I don't actually understand what is the difference between this method and `execute(with value: T)`. Could you please clarify?

File: GliaWidgets/Lib/Command/Command.swift (hunk 1 above)
Comment (gersonnoboa) [reply GliaWidgets/Lib/Command/Command.swift:41]: Ah maybe if you are an English speaker it makes more sense. I don't remember dealing with it in my limited experience with C++ so when I read it the first time I was dumbfounded.

File: GliaWidgets/Lib/Command/Command.swift (hunk 2 above)
Comment (igorkravchenko) [reply GliaWidgets/Lib/Command/Command.swift:37]: Sure: it allows code like this:
```
let validateEmail = Command<String> { email in /* email validation goes here */ }
validateEmail("email@example.test")
```
It makes usage more ergonomic by omitting the call to `execute`. Without it the code would be like this:
```
validateEmail.execute(with: "email@example.test")
```

**Comment by EgorovEI** [comment]
Would be beneficial to add few additional things:
- default values. Something like 
```swift 
static var nop: Command {
    return Command { _ in }
}
```
and 
```swift 
static var nop: Cmd {
    return Cmd { }
}
```
  - Dispatching on queue:
  ```swift
  public func dispatched(on queue: DispatchQueue) -> CommandWith {
    return Command { value in
      queue.async {
        self.execute(with: value)
      }
    }
  }
  ```
  - Chaining:
  ```swift 
  func then(_ another: Command) -> Command {
    return Command { value in
      self.execute(with: value)
      another.execute(with: value)
    }
  }
  ```

**Comment by igorkravchenko** [comment]
> Would be beneficial to add few additional things:
> 
> * default values. Something like
> 
> ```swift
> static var nop: Command {
>     return Command { _ in }
> }
> ```
> 
> and
> 
> ```swift
> static var nop: Cmd {
>     return Cmd { }
> }
> ```
> 
>  * Dispatching on queue:
> 
> ```swift
> public func dispatched(on queue: DispatchQueue) -> CommandWith {
>   return Command { value in
>     queue.async {
>       self.execute(with: value)
>     }
>   }
> }
> ```
> 
>  * Chaining:
> 
> ```swift
> func then(_ another: Command) -> Command {
>   return Command { value in
>     self.execute(with: value)
>     another.execute(with: value)
>   }
> }
> ```

Sure. I'll add those as soon as those are to be used. Also I'd like to see how far I can get without usage of `.nop`.

File: GliaWidgets/Lib/Command/Command.swift (hunk 2 above)
Comment (EgorovEI) [reply GliaWidgets/Lib/Command/Command.swift:37]: Ohh, I see. I totally forgot this `callAsFunction` feature)

**Review by EgorovEI — APPROVED** [review]

**Comment by igorkravchenko** [comment]
!merge

**Pull request merged**